src/controller/submitCode.js

```diff
@@ -115,19 +121,28 @@ async function sendCode(source_code_path, user_id, prob_name) {
             judger = kon.judgers[judgerNum];
         }
 
-        judger.send(source_code_path, prob_name, sub_id).catch((err) => {
-            Console.log(err.message);
-        });
-
-        setImmediate(() => reloadSubs());
-        // Second trigger: 45 seconds
-        setTimeout(() => reloadSubs(), 45000);
+        judger
+            .send(source_code_path, prob_name, sub_id)
+            .then((res) => {
+                if (res.status !== 200) throw Error("Cannot get from Kon");
+            })
+            .then(() => {
+                // Attempt & retry 3 times to get submission
+                for (let i = 1; i < 4; i++)
+                    setTimeout(reloadSub, i * 30000, judger);
```

Comment:
Does this mean the `reloadSub` function is **always** called after 30, 60 and 90 seconds? Even if they succeed from the first time?

Reply:
Yes, this is a trial to test if Wafter & Kon are able to hold. Even if they succeed before, submission's result won't be re-updated